### PR TITLE
[Libretro] Update core options to v2 + add a few new ones

### LINKED ISSUE #15306: Cannot set PSP Model on RetroArch, Toukiden does not work properly~！
### Game or games this happens in

ULJM06290 - Toukiden 

### What area of the game / PPSSPP

QL4(8IRD)7}FKJWLS@E{QTU

### What should happen

Hope to be able to set PSP Model

### Logs

_No response_

### Platform

libretro / Retroarch

### Mobile phone model or graphics card

Pixel 4a

### PPSSPP version affected

RetroArch 1.9

### Last working version

_No response_

### Graphics backend (3D API)

OpenGL / GLES

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

### PR DESCRIPTION
**Pull request by bslenul**
Updated core options to v2 (so we can have categories and descriptions) + added the following options: "Cache Full ISO in RAM", "PSP Model" and "Hardware Tesselation". A review would be welcome! Still learning C/C++ and not used to big PRs like this, so please be gentle! 😳 

For now I only made 3 categories: "System", "Video" and "Network":

image

Still better than having everything mixed IMO. Only put a few descriptions (grabbed from standalone), feel free to add more, I kinda suck at writing in English and I also don't even know what some of these options do, so... :p 

P.S: the "PSP Model" option should fix #15306, this is what I get with the core option set to "PSP-1000":

and when set to "PSP-2000/3000":

Also, Libretro builds seem to be broken currently for Windows and OSX because of a few `unresolved external symbol zip_XXXX referenced`: https://git.libretro.com/libretro/ppsspp/-/jobs/2706096

### CODE REVIEW

**Comment by LibretroAdmin** [comment]
Hi guys, for further clarification, it seemed libretro cores stopped building after this PR was merged -
https://git.libretro.com/libretro/ppsspp/-/commit/3f33cf102c7b33e4c53d3ad9c73c0f0e0aa58a59

Could that possibly be looked into? Thanks.

**Comment by hrydgard** [comment]
@LibretroAdmin sure. Weird one though... Would have expected that we had enough uses of libzip already, that some extended use wouldn't have a risk of causing linker errors..

File: libretro/libretro_core_options.h [hunk 1]
```diff
@@ -0,0 +1,1359 @@
...
+         { "JIT",         "Dynarec (JIT)" },
+         { "IR JIT",      "IR Interpreter" },
+         { "Interpreter", NULL },
+         { NULL, NULL },
+      },
+      "JIT"
+   },
+   {
+      "ppsspp_fast_memory",
+      "Fast Memory (Speedhack)",
```
Comment (unknownbrackets) [inline libretro/libretro_core_options.h:157]: Fast memory isn't a speedhack, it's something of the reverse - having it off is a "slowhack."  The actual PSP always has fast memory enabled.  However, if a bug occurs with fast memory enabled, you get a crash.  Some games (especially with adhoc, etc.) still run into emulator bugs, but if you force the game not to crash... it still mostly works.

That's what this setting does - when disabled, it forces the game not to crash when a PSP playing the game would've crashed (i.e. exiting to the XMB or powering itself off.)

I realize you seem to be just copying what some libretro person decided to call it from the other file.

-[Unknown]

**Comment by unknownbrackets** [comment]
See #16322.  I'm still not interested enough in libretro to test it or figure out how to compile it, obviously the Actions build that someone contributed isn't telling us anything.

Actions is saying to use `#include "libretro.h"`, which it seems other files like libretro_vulkan.h already do.

-[Unknown]

File: libretro/libretro_core_options.h (hunk 1 above)
Comment (bslenul) [reply libretro/libretro_core_options.h]: Thank you for the explanation! And yes I indeed copied most of the options as-is, "(Speedhack)" removed :)

**Comment by bslenul** [comment]
Thank you for the build fix, building just fine now! :+1: 

> or figure out how to compile it

On Windows (using MinGW64) I have to `export VsInstallRoot="/path/to/Microsoft Visual Studio/2019/Community"` then `make -j4 platform=windows_msvc2019_desktop_x64` (from libretro dir).
On Linux it's a bit easier, I just "make" from the libretro dir.
CMake should work as well, I do this from a build dir personally:
```
cmake .. -DLIBRETRO=ON -DCMAKE_BUILD_TYPE=Release
cmake --build . --target ppsspp_libretro --config Release
```

**Comment by LibretroAdmin** [comment]
Looks fine to me for whatever it's worth.

**Review by hrydgard — APPROVED** [review]

**Comment by hrydgard** [comment]
Still some build errors though, as unknown says, please replace 

`#include <libretro.h>`

with

`#include "libretro.h"` or with a more detailed path if appropriate.

**Comment by bslenul** [comment]
Oh sorry, completely missed it, let's try again!

**Pull request merged**